pkg/daemon/ceph/osd/daemon.go

```diff
@@ -37,6 +38,33 @@ var (
 	logger = capnslog.NewPackageLogger("github.com/rook/rook", "cephosd")
 )
 
+// StartOSD starts an OSD on a device that was provisioned by ceph-volume
+func StartOSD(context *clusterd.Context, bluestore bool, osdID, osdUUID string, cephArgs []string) error {
```

Comment:
Couldn't the operator create separate init containers to (1) ensure the mount point exists and (2) execute the `ceph-volume lvm activate` command , and then run `ceph-osd ...args` in the main container?

Reply:
`ceph-volume lvm activate` creates a `tmpfs` mount that is the same as the lifetime of the container. So if we used an init container, the `activate` mount would effectively disappear when the main container is launched with ceph-osd

Reply:
That's a bit of an annoyance, but thanks for the info. That might be be a good code comment for when I forget later if you're still making changes.

Reply:
yes, i'll add a comment for that, thanks

Reply:
@travisn @BlaineEXE Is a shared `tmpfs` "needed"? If we need one, I think we would get a `tmpfs` shared between containers in the Pod using `emptyDir` with `medium: memory`.

Reply:
Or you can just copy the files from the tmpfs mount to a shared mount between the init container and the `ceph-osd` one. 🙂

Reply:
I don't think that you can just copy `/var/lib/ceph/osd/<tmpfs>/*` contents like that. In the case of bluestore, things aren't even mounted, just symlinked to the devices themselves

Reply:
i just completed an experiment and verified that the contents of the tmpfs will **not** survive between an init container and a ceph-osd container. The only way I see to get `ceph-osd` in its own container is by copying the config. But agreed with @alfredodeza that it's more than a simple copy because of the symlinks and I don't see that it is worth the risk. 

For 0.9 i'll stick with the implementation as currently in the PR by launching `rook`, which then creates processes for `ceph-volume` to activate, then `ceph-osd` for the daemon. If we find a better alternative we can change it after 0.9.

Reply:
@alfredodeza @travisn Worked for me, as documented in https://github.com/rook/rook/issues/1342#issuecomment-412842152 .